Drow and the Underdark

CometBlaze opened the pull request:
## About The Pull Request

<!-- Describe your pull request. Avoid text walls, use concise bullet points for easier readability. Document every change, or this can delay review and even discourage maintainers from merging your PR. -->
Added a new unique Drow armour called "Drowcraft Vest" with the same stats as heavy leather armour. Currently only found being worn by Drow NPCs.
Drow NPCs will now favour dodging instead of parrying to further differentiate them from other NPCs and to align more closely with the expected Drow fighting style, fast and agile killers.
Added a new weapon: the Stalker Falx, a Drow variant of the falx which trades damage for swift balance, and gave it to Drow NPCs.
Drow NPCs can now spawn with a whip, the new falx or even dual weilding daggers.
Lastly, Drow Mercenaries (Black Venom and Anthrax Assassins) now get an area buff when in the Underdark, giving them +3 speed and +1 perception.

As an aside, this PR was also intended to add a fast travel point that connected the Underdark to the surface to allow for it to see some more use as it's currently very out of the way and makes things like using the undercities as a base of operations hard to do. The code side of this is also done in this PR and a follow-up PR will be made to add in the mapping changes once some of the backlog has been cleared.
## Testing Evidence

<!-- It's mandatory to test your PR. Provide images, clips or description of how you tested your changes where possible. -->
<img width="1085" height="1080" alt="image" src="https://github.com/user-attachments/assets/c13c07b9-c113-4ba3-bce3-6e28b975aa8b" />
<img width="1092" height="1080" alt="image" src="https://github.com/user-attachments/assets/7463fb5e-1ef8-4af3-bf9b-889201faa142" />
<img width="485" height="140" alt="image" src="https://github.com/user-attachments/assets/8f029223-e0a7-4e04-8560-d5f37caf8caa" />
<img width="187" height="191" alt="image" src="https://github.com/user-attachments/assets/e5d2511c-f3b6-496a-aa1e-2a25dc6eaec9" />

## Why It's Good For The Game
It annoyed me to no end how the supposedly terrifying Drow of the Underdark looked like generic mooks in armour, it cheapened the feel and made them rather generic. Changing how they look and how they fight will go a long way towards making them feel more unique.
The area buff is meant to further reinforce the fact that finding a Drow in the Underdark is really scary and to encourage Drow players to actually venture into their turf and act as the mushroom lorax for a little bit.
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial. If you can't, then it probably isn't good for the game in the first place. -->

<!-- By contributing to this codebase, you confirm that any code and sprites you provide are legal to share and will be licensed under the terms specified in README.md — AGPLv3 for code and CC-BY-SA 3.0 for assets, unless otherwise stated. You acknowledge that the project maintainers are under no obligation to remove any materials that do not violate these licenses. -->

## Changelog
<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

<!-- !! Do not add whitespace in-between the entries, do not change the tags and do not leave the tags without any entries. -->

:cl:
add: Added a new weapon: the Stalker Falx, a Drow variant of the falx
add: Added a new armour: the Drowcraft Vest. Currently only available to Drow NPCs
balance: Drow raider NPCs now dodge instead of parrying
balance: Black Venom and Anthrax Assassin now get a stat buff while in the Underdark
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

CometBlaze commented:
Bear with me as I try to unfuck the mapping portion that got through unintentionally

DrunkenLombax commented:
While I FW this heavy, I have to disagree with the decision to give them merc armour (even if "weakened"), considering the drow in our underdark area arent supposed 2 be part of Black Venom or Anthraxi assassins (which are both their own drow merc guilds, as will be apparent in the underdark lore I've prepared to come soon). I feel like visually the merc gear should stay primarily locked to mercenaries, lest their fit lose its importance.

Everything else is a big thumbs up from me, but I think them keeping the iron gear is fine? Unless someone drops some sprites and fits for the drow npc's specifically, which is unlikely.

HeroWithYay commented:
Its peak

CometBlaze commented:
PR is ready for review.

Added an unique Drow armour with the same stats as heavy leather and replaced the merc gear with it for NPCs (pictures in the PR description)

Mapping changes still left out until the backlog of mapping changes is cleared because I don't want to deconflict the same PR a bunch of times.

CometBlaze commented:
> +3 speed feels a little excessive bearing in mind Anthrax Assassins already get +2 innately?

It might be, I didn't think terribly hard about the values, just grabbed the warden buff and flipped it around. But given this only applies to such a limited area that doesn't see that much travel I'm not sure it will be too much of a problem. I'm not opposed to bumping it down if it's needed.
